File: resources/vue/src/api/plugins.ts
import request from '@/utils/request'

var api = '/api/plugins/'

export function PluginMarket(data) {
  return request({
    url: api + 'PluginMarket',
    method: 'post',
    data
  })
}

export function GetPluginInfo(data) {
  return request({
    url: api + 'GetPluginInfo',
    method: 'post',
    data
  })
}

export function InstallPlugin(data) {
  return request({
    url: api + 'InstallPlugin',
    method: 'post',
    data
  })
}

export function UploadPlugin(data) {
  return request({
    url: api + 'UploadPlugin',
    method: 'post',
    data
  })
}

export function StarPlugin(data) {
  return request({
    url: api + 'StarPlugin',
    method: 'post',
    data
  })
}

export function ImportEvKey(data) {
  return request({
    url: api + 'ImportEvKey',
    method: 'post',
    data
  })
}

export function UnInstallPlugin(data) {
  return request({
    url: api + 'UnInstallPlugin',
    method: 'post',
    data
  })
}

export function LikeComment(data) {
  return request({
    url: api + 'LikeComment',
    method: 'post', 
    data
  })
}

 
export function ListComments(data) {
  return request({
    url: api + 'ListComments',
    method: 'post', 
    data
  })
}

 
export function AddComment(data) {
  return request({
    url: api + 'AddComment',
    method: 'post', 
    data
  })
}

export function GetWxArticleList(data) {
  return request({
    url: api + 'GetWxArticleList',
    method: 'post',
    data
  })
}

export function GetLocalPluginList(data) {
  return request({
    url: api + 'GetLocalPluginList',
    method: 'post',
    data
  })
}

File: pkg/services/web/plugins.go
package web

// ES基础操作 路由
func (this *WebServer) runPlugins() {

	const AbsolutePath = "/api/plugins"
	group := this.engine.Group("插件操作", AbsolutePath)
	{
		group.POST(true, "离线上传插件", "/UploadPlugin", this.pluginController.UploadPlugin)
		group.Use(this.middleWareService.OperaterLog)
		group.POST(false, "查询本地插件列表", "/GetLocalPluginList", this.pluginController.GetLocalPluginList)
		group.POST(false, "导入EvKey到配置文件", "/ImportEvKey", this.pluginController.ImportEvKey)
		group.POST(false, "查询插件动态", "/GetWxArticleList", this.pluginController.GetWxArticleList)
		group.POST(false, "查询插件市场", "/PluginMarket", this.pluginController.PluginMarket)
		group.POST(false, "查询插件详情", "/GetPluginInfo", this.pluginController.GetPluginInfo)
		group.POST(true, "安装插件", "/InstallPlugin", this.pluginController.InstallPlugin)
		group.POST(true, "卸载插件", "/UnInstallPlugin", this.pluginController.UnInstallPlugin)
		group.POST(false, "给插件star", "/StarPlugin", this.pluginController.StarPlugin)
		group.POST(false, "给插件评论", "/AddComment", this.pluginController.AddComment)
		group.POST(false, "给插件评论点赞", "/LikeComment", this.pluginController.LikeComment)
		group.POST(false, "给插件star", "/ListComments", this.pluginController.ListComments)

	}
}
